tbmalt/common/maths/interpolation.py

```diff
@@ -237,6 +241,9 @@ class PolyInterpU:
 
     """
 
+    # TODO: Resolve bug causing incorrect value to be returned when
```

Comment:
Hi Adam, is there an example or unittest to reproduce this bug? I will also have a look on this issue.

Reply:
Although this example is rather contrived it does demonstrate how non-trivial the error can become under certain situations:
```python
from tbmalt.common.maths.interpolation import PolyInterpU
import torch

x = torch.linspace(0, 10, 10)
y_reference = torch.rand(10)
poly = PolyInterpU(x, y_reference)
y_predicted = poly(x)
y_delta = y_predicted - y_reference
```
The error becomes most pronounced at location of the last data-point. In this example the predicted value can be up to ±60 off from the reference value which is only ever in the range 0.0-1.0.